group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -8148,7 +8174,10 @@ private TaskAssignor streamsGroupAssignor(String groupId) {
      * Get the assignor of the provided streams group.
      */
     private Map<String, String> streamsGroupAssignmentConfigs(String groupId) {
```

Comment:
For my own education: why does this method return a `Map`? In the end, we _know_ the config name, and I would assume the caller wants to lookup the config value. Wo why not just return an `int` ?

Reply:
This is just prepared to have more assignment configs - e.g. `max_warmup_replicas`. In the future, there will be more assignment configs that we need to pass around. We could alternatively just have a little class (`StreamsAssignmentConfigs` or something). Having a simple map would make it easier adding a pluggable assignor interface later on, since there is less worrying about compatibility with primitive types. But if you prefer, we can add a container with names.

Reply:
Thanks. Should be good for now. It's all internal stuff so we can still refactor in the future is we want to.